### PR TITLE
Persist the web app's storage folder

### LINKED ISSUE #42: Adding a Volume to a Laravel Web App
Hi Team! I'm planning on  adding an option to add volume to the Laravel Fly app:

**Question:**
Should we have a separate command for adding volume? Or should it be included in the launch command?
-I'm thinking it should be part of the launch command since it's not a separate app from the laravel web app, unlike mysql, redis, etc( which would be on a separate command right?)

_______________________________________________________________________________________________________________

**Steps:**
-The user will be prompted if they want to add a volume to the Laravel Fly App, 
-Prompt for directory to attach the volume, with default being the storage folder
-Prompt for size of volume to attach, not sure about default size yet

### PR DESCRIPTION
**Pull request by KTanAug21**
Create a volume service class:
1. Prompts whether user wants to persist the storage folder
2. Includes mount in the fly.toml array
3. Creates a .fly script to restore storage content ( since mounting a volume to the folder initially erases its content )
4. Backs up storage to storage_ folder( used in the .fly script to restore storage folder )
5. Deletes storage_ folder locally during deploy

### CODE REVIEW

File: app/Commands/LaunchCommand.php [hunk 1]
```diff
@@ -64,7 +65,14 @@ public function handle(FlyIoService $flyIoService)
                 return true;
             });
 
-            // 3. Detect Node and PHP versions
+            // 3. Ask user whether to mount volume to storage folder
+            $mount = [];
+            $volumeService = new VolumeService($this);
+            $this->task("Determine whether to persist storage folder", function() use(&$mount, $volumeService, $appName){
+                $mount = $volumeService->promptVolume( $appName );
```
Comment (Johannes-Werbrouck) [inline app/Commands/LaunchCommand.php:72]: The main branch has seen a lot of changes, one of them is that the handle method is split up in an `input...` and a `setUp...` method for every LaunchCommand. This way, all the user input can be consolidated in the `input...` method, and used later on to set everything up in the `setUp...` method. Could you please follow this logic here as well?

File: app/Services/VolumeService.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        ];
+        return $this;
+    }
+    
+    // Prompt User
+    public function promptVolume( 
+        string $laravelAppName
+    ){
+        $volumeMount = [];
+        $answer = $this->command->choice(
```
Comment (Johannes-Werbrouck) [inline app/Services/VolumeService.php:27]: you could use `$this->command->confirm()` here instead of `->choice()`

**Review by Johannes-Werbrouck — CHANGES_REQUESTED** [review]
A lot has changed in the main branch, sorry for that !😅 
Please merge the main branch into your branch and check my comments, some small changes will be needed. Thanks!

**Comment by KTanAug21** [comment]
> A lot has changed in the main branch, sorry for that !😅 Please merge the main branch into your branch and check my comments, some small changes will be needed. Thanks!

You've done an amazing job cleaning and adding new features to fly-laravel! ✨No worries on that! 

Also, I'm planning on adding tests to existing features in the fly-laravel package, so I'm not sure yet whether I'll finalize this PR or those tests first. Don't know which one I'll work on first, but I'll just finalize an article for this week and get back to the fly-laravel package.

Awesome job so far @Johannes-Werbrouck!

**Comment by fideloper** [comment]
Another thing someone mentioned to me in a youtube comment is that some people symlink `public/storage` to `storage/app/public` (or similar), which works with the [`storage:link`](https://laravel.com/docs/10.x/filesystem#the-public-disk) command.

I wonder if the `storage:link` command is one we could add in the `.fly/scripts` command as well.

(I don't think there are negative consequences to assuming this command can be run on every deploy, but LMK if you think of any)

**Pull request closed without merging**